internal/retry/schema.go

```diff
@@ -0,0 +1,84 @@
+package retry
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/Azure/terraform-provider-azapi/internal/services/myvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/numberdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+const (
+	intervalSecondsAttributeName     = "interval_seconds"
+	maxIntervalSecondsAttributeName  = "max_interval_seconds"
+	multiplierAttributeName          = "multiplier"
+	randomizationFactorAttributeName = "randomization_factor"
+	errorMessageRegexAttributeName   = "error_message_regex"
+)
+
+func SingleNestedAttribute(ctx context.Context) schema.Attribute {
```

Comment:
How about renaming the block to `retry`? Because maybe in the future we could introduce other customized retry logics like retry untils the resource's status becomes X.

```
retry {
    error_message_regex = [ "ResourceGroupNotFound" ]
    multiplier = 1.5
    randomization_factor = 0.5
    interval_seconds = 10
    max_interval_seconds = 180
}
```

Reply:
Refactored in: 4f6a1361e1b83f1e51f6524f70ad2a2515ba2ce8